main.go

```diff
@@ -218,8 +330,8 @@ func main() {
 		}
 		defer tconn.Close()
 
-		tl := &StatsDTCPListener{conn: tconn, eventHandler: eventQueue, logger: logger}
-		go tl.Listen()
+		tl := &listener.StatsDTCPListener{Conn: tconn, EventHandler: eventQueue, Logger: logger}
+		go tl.Listen(linesReceived, eventsFlushed, tcpConnections, tcpErrors, tcpLineTooLong, *sampleErrors, samplesReceived, tagErrors, tagsReceived)
```

Comment:
Stylistic question: is it a common pattern to pass the counters like this? It seems a bit prone to accidentally messing up the order and counting wrongly … would it make sense to make them fields in the StatsdDTCPListener struct?

Reply:
Probably! Let me know if everything else looks ok and I can try to get that in there... I'm sure there's a lot of sub-optimal stuff as I'm just getting into the code!